### DISCUSSION TITLE
Route Firebase Notification to sms app interval

### DISCUSSION THREAD

**Discussion by cilinjohnrey** [ibnux/Android-SMS-Gateway#19] [Q&A] [answered]
Hi author, I try to modify the code to increase the interval of each sms message to route in sms app. I would like to know where is the function to route the sms message to sms app so that i can put random interval 16 to 30 seconds to it.

**Comment by ibnux** [comment, marked as answer]
The function to send SMS are in this area

https://github.com/ibnux/Android-SMS-Gateway/blob/a7fcfddf88f79f983b5956ecb760eb6132b83e71/app/src/main/java/com/ibnux/smsgateway/layanan/PushService.java#L261-L272

that go to SimUtil if multiple SIMCARD

https://github.com/ibnux/Android-SMS-Gateway/blob/master/app/src/main/java/com/ibnux/smsgateway/Utils/SimUtil.java

or to Fungsi (function) if just 1 simcard

https://github.com/ibnux/Android-SMS-Gateway/blob/a7fcfddf88f79f983b5956ecb760eb6132b83e71/app/src/main/java/com/ibnux/smsgateway/Utils/Fungsi.java#L43-L98
